landingai/vision_pipeline.py

```diff
@@ -204,6 +210,49 @@ def save_image(self, filename_prefix: str, image_src: str = "") -> "FrameSet":
             c += 1
         return self
 
+    def save_video(
```

Comment:
1. please add unit tests for it, cover a few cases where `video_fps` is none versus not none, `video_length_sec` is none versus not none.
2. how about create a low level API first, i.e. in a dedicated video module, then create wrapper in vision pipeline.

Reply:
Unit test added. Layering I prefer to delay it until we need it

Reply:
Wouldn't it be better if we reverse the priority?
Prioritize lower level function over the higher layer, we can defer introducing the higher layer if the bandwidth is limited. The foundation (lower level function) is less likely to be changed than higher level functions.

This could be generalized as a rule of thumb for future function/cases, maybe worth an in-person discussion.

Reply:
Add the topic to our discussion list